streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -244,18 +274,117 @@ public static Map<ProcessId, KafkaStreamsAssignment> optimizeRackAwareStandbyTas
+        final BiFunction<KafkaStreamsAssignment, KafkaStreamsAssignment, List<TaskId>> getMovableTasks = (source, destination) -> {
+            return source.tasks().values().stream().filter(task -> task.type() == AssignedTask.Type.STANDBY)
+                .filter(task -> {
+                    return !destination.tasks().containsKey(task.id());
+                })
+                .filter(task -> {
+                    final KafkaStreamsState sourceState = kafkaStreamsStates.get(source.processId());
+                    final KafkaStreamsState destinationState = kafkaStreamsStates.get(source.processId());
+                    return moveablePredicate.canMoveStandbyTask(sourceState, destinationState, task.id(), kafkaStreamsAssignments);
+                })
+                .map(AssignedTask::id)
+                .sorted()
+                .collect(Collectors.toList());
+        };
+
+        final long startTime = System.currentTimeMillis();
+        boolean taskMoved = true;
+        int round = 0;
+        final RackAwareGraphConstructor<KafkaStreamsAssignment> graphConstructor = RackAwareGraphConstructorFactory.create(
+            applicationState.assignmentConfigs().rackAwareAssignmentStrategy(), taskIds);
+        while (taskMoved && round < STANDBY_OPTIMIZER_MAX_ITERATION) {
+            taskMoved = false;
+            round++;
+            for (int i = 0; i < kafkaStreamsAssignments.size(); i++) {
+                final UUID clientId1 = clientIds.get(i);
+                final KafkaStreamsAssignment clientState1 = kafkaStreamsAssignments.get(new ProcessId(clientId1));
+                for (int j = i + 1; j < kafkaStreamsAssignments.size(); j++) {
+                    final UUID clientId2 = clientIds.get(i);
+                    final KafkaStreamsAssignment clientState2 = kafkaStreamsAssignments.get(new ProcessId(clientId2));
+
+                    final String rack1 = clientRacks.get(clientState1.processId().id()).get();
+                    final String rack2 = clientRacks.get(clientState2.processId().id()).get();
+                    // Cross rack traffic can not be reduced if racks are the same
+                    if (rack1.equals(rack2)) {
+                        continue;
+                    }
+
+                    final List<TaskId> movable1 = getMovableTasks.apply(clientState1, clientState2);
+                    final List<TaskId> movable2 = getMovableTasks.apply(clientState2, clientState1);
+
+                    // There's no needed to optimize if one is empty because the optimization
+                    // can only swap tasks to keep the client's load balanced
+                    if (movable1.isEmpty() || movable2.isEmpty()) {
+                        continue;
+                    }
+
+                    final List<TaskId> taskIdList = Stream.concat(movable1.stream(),
+                            movable2.stream())
```

Comment:
I noticed this is messed up in the original code too, but can you fix the indentation/remove the line break (just here, no need to fix it in the RackAwareTaskAssignor too)